spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/AbstractConnectionFactory.java

```diff
@@ -212,7 +212,7 @@ protected ApplicationEventPublisher getApplicationEventPublisher() {
 
 	@Override
 	public void onApplicationEvent(ContextClosedEvent event) {
-		if (getApplicationContext().equals(event.getApplicationContext())) {
+		if (getApplicationContext() != null && getApplicationContext().equals(event.getApplicationContext())) {
```

Comment:
Hi @garyrussell,
Thanks for checking this issue.
I of course agree with you preferred approach of the `if` statement.
Unfortunately this issue occurs in quite a big SpringBoot application and I need to see whether I do get the time to create a small example demonstrating the issue.
It might be possible / easier to analyse why `setApplicationContext` doesn't get called (or called with `applicationContext` being `null`).
Kind regards!